bugfix/nfcens_rtofs_R2

SamiraArdani-NOAA opened the pull request:
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>
This PR addresses NCO comment regarding the use of cp -v instead of cpreq in EVS-NFCENS and EVS-RTOFS  

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

1- Clone my fork in your local and checkout branch "bugfix/nfcens_rtofs_R2".

2- Testing EVS-NFCENS:
- Only run the plots STEP for this change.
- To run for plots, find the driver scripts located at: EVS/dev/drivers/scripts/plots/nfcens/jevs_nfcens_wave_grid2obs_plots.sh

3- Testing EVS-RTOFS:
Please test it for all the STEPS: prep, stats and plots
- for prep: EVS/dev/drivers/scripts/prep/rtofs/jevs_rtofs_prep.sh
- for stats: EVS/dev/drivers/scripts/stats/rtofs/*
- for plots: EVS/dev/driver/scripts/plots/rtofs/*

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

SamiraArdani-NOAA committed:
Mail was wrapped.

SamiraArdani-NOAA committed:
Minor typo was fixed.

SamiraArdani-NOAA committed:
SENDMAIL was set.

SamiraArdani-NOAA committed:
Merge branch 'develop' into bugfix/nfcens_R2

SamiraArdani-NOAA committed:
Merge branch 'develop' into bugfix/nfcens_rtofs_R2

SamiraArdani-NOAA committed:
cpreq was replaced with cp -v.

SamiraArdani-NOAA committed:
Merge branch 'develop' into bugfix/nfcens_rtofs_R2

SamiraArdani-NOAA committed:
Bug was fixed for copying tar files.

SamiraArdani-NOAA committed:
Merge branch 'develop' into bugfix/nfcens_rtofs_R2

AliciaBentley-NOAA commented:
@PerryShafran-NOAA @ShelleyMelchior-NOAA I have looked over this PR and believe that it looks good! Once tested, I approve this for being merged. Thank you, @SamiraArdani-NOAA!

PerryShafran-NOAA commented:
@SamiraArdani-NOAA I am starting work on this PR.  I have started with the nfcens plot job.  The .o file will be located here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr439test/EVS/dev/drivers/scripts/plots/nfcens

PerryShafran-NOAA commented:
@SamiraArdani-NOAA The plot job is already finished.  Please look at the .o file, and also look at the output directory here:  /lfs/h2/emc/ptmp/perry.shafran/evs/v1.0/plots/nfcens/wave.20240206

PerryShafran-NOAA commented:
@SamiraArdani-NOAA I have also started the rtofs prep job.  The .o file will be here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr439test/EVS/dev/drivers/scripts/prep/rtofs

PerryShafran-NOAA commented:
@SamiraArdani-NOAA I also started the rtofs stats jobs.  You can see all the .o files here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr439test/EVS/dev/drivers/scripts/stats/rtofs

SamiraArdani-NOAA commented:
@PerryShafran-NOAA : the NFCENS plots .o file looks good. The plot .tar files are identical to those in with emc.vppppg.

SamiraArdani-NOAA commented:
@PerryShafran-NOAA : If you have not started, would you please run the plot jobs for 20240204 (instead of 20240205)? The outputs have been populated only up to 20240204 in emc.vpppg.

PerryShafran-NOAA commented:
@SamiraArdani-NOAA To be clear, you are referring to the rtofs plot jobs?

PerryShafran-NOAA commented:
@SamiraArdani-NOAA Should I also run the stats jobs at 20240204 as well?  I just realized that these jobs are usually run at 22Z and may need to restart.

SamiraArdani-NOAA commented:
@PerryShafran-NOAA : Yes. That is right, for only rtofs.

SamiraArdani-NOAA commented:
@PerryShafran-NOAA : yes, please. It would be better to have something in emc.vpppg for comparison.

PerryShafran-NOAA commented:
Stats runs were restarted for VDATE=20240204.

SamiraArdani-NOAA commented:
@PerryShafran-NOAA : Thank you.

AliciaBentley-NOAA commented:
@SamiraArdani-NOAA These RTOFS jobs take a while to complete, right?

SamiraArdani-NOAA commented:
@AliciaBentley-NOAA : Yes. it takes a while. I am still waiting to see the stats for argo and ndbc.

PerryShafran-NOAA commented:
@SamiraArdani-NOAA Can you report on the results for everything but argo and ndbc?  The other stats jobs should be done.  

You should be able to see results here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/rtofs
The .o files are located in the directory I indicated above.

SamiraArdani-NOAA commented:
@PerryShafran-NOAA : stat .o files for runs smap, smos, ghrsst, aviso and osisaf look good. The results matched with those in emc.vpppg.

SamiraArdani-NOAA commented:
@PerryShafran-NOAA : stat results for run case argo look good.

SamiraArdani-NOAA commented:
@PerryShafran-NOAA , @AliciaBentley-NOAA : Do you agree that I include addressing comment no. 86 for RTOFS- prep here? We can save time since RTOFS prep steps takes several hours. I also asked @ShelleyMelchior-NOAA to do this so she also knows.

AliciaBentley-NOAA commented:
@SamiraArdani-NOAA Would you need to update this PR and we would need to run tests again? I'm fine with that, since it's already open and would require us to run fewer tests overall. If you do that, please update your initial submission to include the item 86 topics that this PR also covers (and how to test them). Thanks!

@PerryShafran-NOAA

SamiraArdani-NOAA commented:
@AliciaBentley-NOAA : it is just for RTOFS-prep (only one test) and it will be fixed easily by adding WARNING messages outside $SENDMAIL. I have already done with changes today. Just need to commit here.

AliciaBentley-NOAA commented:
@SamiraArdani-NOAA Yes, I think that's perfectly reasonable! Commit them here and let us know when Perry should re-run RTOFS prep. We'll probably have SENDMAIL set to NO for the test as well, to make sure that block isn't even executed. @PerryShafran-NOAA

SamiraArdani-NOAA committed:
Addressed comment no. 86 for prep job.

PerryShafran-NOAA commented:
@SamiraArdani-NOAA @AliciaBentley-NOAA Changes pulled in and new run of rtofs prep has been started.  You can check out the file jevs_rtofs_prep.o132137757 for progress.

PerryShafran-NOAA commented on ush/rtofs/rtofs_prep_regions.sh:
```diff
@@ -80,7 +88,9 @@ if [ ! -s $COMOUTprep/rtofs.$VDATE/$RUN/south_atlantic.lat.nc ]; then
         if [ $SENDCOM = "YES" ]; then
-            cpreq -v $DATA/rtofs.$VDATE/$RUN/south_atlantic.lat.nc $COMOUTprep/rtofs.$VDATE/$RUN
+		if [-s $DATA/rtofs.$VDATE/$RUN/south_atlantic.lat.nc ]; then
+            		cp -v $DATA/rtofs.$VDATE/$RUN/south_atlantic.lat.nc $COMOUTprep/rtofs.$VDATE/$RUN
```
@SamiraArdani-NOAA 

Line 91 needs a space between the bracket and the dash.  Otherwise we get this error:

`/lfs/h2/emc/vpppg/noscrub/perry.shafran/pr439test/EVS/ush/rtofs/rtofs_prep_regions.sh: line 91: [-s: command not found`

Please correct this.

SamiraArdani-NOAA replied:
@PerryShafran-NOAA : oops. My bad. I will fix it. Thanks.

PerryShafran-NOAA commented:
@SamiraArdani-NOAA I made a fix locally to that ush script and am rerunning.  That rerun would be jevs_rtofs_prep.o132139367. You'll still have to make the change in your fork to fix that syntax.

SamiraArdani-NOAA committed:
Minor bug was fixed.

PerryShafran-NOAA commented:
@SamiraArdani-NOAA I also started the plot jobs - you can see the .o files here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr439test/EVS/dev/drivers/scripts/plots/rtofs

PerryShafran-NOAA commented:
@SamiraArdani-NOAA The plot runs are finished.  Please see the .o files here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr439test/EVS/dev/drivers/scripts/plots/rtofs

And the output tarballs are here:
/lfs/h2/emc/ptmp/perry.shafran/evs/v1.0/plots/rtofs

Please look at both the rtofs and headline directories.

SamiraArdani-NOAA commented:
@PerryShafran-NOAA : all plots and headline are identical with emc.vpppg. I am still waiting for argo plots.

PerryShafran-NOAA commented:
@SamiraArdani-NOAA Please do a sync fork.

SamiraArdani-NOAA commented:
@PerryShafran-NOAA : the argo plots also look good.

SamiraArdani-NOAA committed:
Merge branch 'develop' into bugfix/nfcens_rtofs_R2

PerryShafran-NOAA commented:
@SamiraArdani-NOAA I think the ndbc stats job is now complete; would you take a look at the output?

I think all that's outstanding is the rtofs prep job?

SamiraArdani-NOAA commented:
@PerryShafran-NOAA : I checked the ndbc stat results. The look identical to emc.vpppg.

SamiraArdani-NOAA commented:
@PerryShafran-NOAA : Yes. Only rtofs prep remains.

SamiraArdani-NOAA commented:
@PerryShafran-NOAA : the rtofs prep job was finished.  The .o file and the results look good.

PerryShafran-NOAA commented:
Thank you!  I will do a quick code check and merge.

PerryShafran-NOAA commented:
@SamiraArdani-NOAA @AliciaBentley-NOAA @ShelleyMelchior-NOAA I just realized that in order to properly test the Wei item 86 changes, I'll need to test by using a fake input directory, to see if those WARNING statements worked as intended.  I need to do another run of the prep job, by using fake input directories.  I'll get that going overnight and hopefully that will work out.

PerryShafran-NOAA commented:
@AliciaBentley-NOAA @SamiraArdani-NOAA Apologies, forgot to set SENDMAIL to NO, so you all have just received a bunch of missing data emails.  I set it to NO and ran again.  

Because this has no data to work with, it was very fast.  Please take a look at the .o file jevs_rtofs_prep.o132165374 and tell me if you see all the WARNING statements that you had added with this PR.

SamiraArdani-NOAA commented:
@PerryShafran-NOAA : The .o file looks as expected. I saw the warning messages for missing observational data and missing rtofs data.

PerryShafran-NOAA commented:
@SamiraArdani-NOAA Excellent!  Now we can go ahead and merge.

PerryShafran-NOAA reviewed (approved):
Code works as expected.  Approved for merge.
